src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -264,11 +264,32 @@ private void RaiseTextChanged()
 					_isTextChangedPending = false;
 				}
 			}
-			
-			_textBoxView?.SetTextNative(Text);
+
+			//Everytime we change the Text property, the cursor turns to 0 / start 
+			var ss = SelectionStart;
+
+			_textBoxView?.SetTextNative(_casingText);
+
+			SelectionStart = ss;
+
 		
 		}
 
+		protected string _casingText
```

Comment:
A few things:
- Does not need to be protected
- The name must use Pascal casing, and no leading underscore
- Indentation on newlines for brackets
- Use the new switch expressions instead of a `case`/`return`

Reply:
Ok!. Internal maybe? As it doen't match the Microsoft Win UI Docs to be public. 
About ToUpper / ToLower, would be better to be Culture Invariant?

Reply:
On a second thought, I think we don't need the CasingText property actually.
Will commit a new proposal, with the Text Property Get method like:

```
public string Text
{
	get => CharacterCasing switch
		{
			CharacterCasing.Lower => ((string)this.GetValue(TextProperty)).ToLower(),
			CharacterCasing.Upper => ((string)this.GetValue(TextProperty)).ToUpper(),
			_ => (string)this.GetValue(TextProperty)
		}; 
	set
	{
		if (value == null)
		{
			throw new ArgumentNullException();
		}

		this.SetValue(TextProperty, value);
	}
}
```
It will be clearer, easier, faster and probably will solve CI tests.
If succeeded, i will squash everything.

Reply:
Oh! :( on TextBox.macOS.cs we have the  _isPassword property. Will not be a good idea :(